### PR TITLE
Web Lab: some logging of loading

### PR DESCRIPTION
**Pull request by breville**
We are seeing issues with Web Lab not properly loading for some students, and are attempting to gather some early data.

This change logs events for the following:

- beginning of load
- Bramble being mountable
- Bramble being ready
- Bramble file system reset failure
- Bramble error
- The "init state" after 20 seconds, making it a bit easier to find sessions where Bramble has failed to load.
- The "reachability" state of an image (https://downloads.computinginthecore.org/favicon.ico) which is on the same domain that hosts the code that we load.  We use an image to avoid CORS restrictions; this image has been added to the S3 bucket for this purpose.
- Also the reachability of another image (https://codeprojects.org/favicon.ico) for another domain we also rely upon.

Some of these were already being logged on New Relic but we don't seem to have access to that data at the moment, possibly because of quota overages.

It would be great to be able to peek in the DOM after 20 seconds, to naively see what has actually been loaded, but this doesn't seem to be possible while we load an iframe from https://downloads.computinginthecore.org/bramble_0.1.30/index.html, unless we have a mechanism to post a message over.  This might be another argument to load Bramble from a subdomain of `code.org` in the near future.

### CODE REVIEW

File: apps/src/weblab/CdoBramble.js [hunk 1]
```diff
@@ -5,6 +5,33 @@ import {FatalErrorType} from './constants';
 import logToCloud from '../logToCloud';
 import {Buffer} from 'buffer';
 
+import testImageAccess from '../code-studio/url_test';
+import firehoseClient from '../lib/util/firehose';
+
+function tempLog(event, data = null) {
+  //console.log(event, data || "");
```
Comment (sanchitmalhotra126) [inline apps/src/weblab/CdoBramble.js]: nit: leftover?

File: apps/src/weblab/CdoBramble.js [hunk 2]
```diff
@@ -5,6 +5,33 @@ import {FatalErrorType} from './constants';
...
+    },
+    {includeUserId: true}
+  );
+}
+
+function testReach() {
+  testImageAccess(
+    'https://downloads.computinginthecore.org/favicon.ico' +
+      '?' +
+      Math.random(),
```
Comment (sanchitmalhotra126) [inline apps/src/weblab/CdoBramble.js:30]: Is the `Math.random()` here to prevent cached URLs? should use `Date.now()` in that case?

**Review by sanchitmalhotra126 — APPROVED** [review]
Seems like a good start! General comment - mind adding some comments on what we're trying to investigate here in case this code ends up sticking around for a while?

File: apps/src/weblab/CdoBramble.js (hunk 2 above)
Comment (breville) [reply apps/src/weblab/CdoBramble.js:30]: Yes, though for better or worse, we seem to use `Math.random()` for most of our `testImageAccess` calls.

Comment (sanchitmalhotra126) [reply apps/src/weblab/CdoBramble.js:30]: Gotcha, yeah not a big deal if we have precedent of using it elsewhere.

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**